localstack/services/apigateway/integration.py

```diff
@@ -518,6 +526,69 @@ def invoke(self, invocation_context: ApiInvocationContext):
         self.response_templates.render(invocation_context)
         return invocation_context.response
 
+    @classmethod
+    def _validate_required_params(cls, request_parameters: Dict[str, Any]) -> None:
+        if not request_parameters:
+            raise BadRequestException("Missing required parameters")
+        # https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-aws-services-reference.html#Kinesis-PutRecord
+        stream_name = request_parameters.get("StreamName")
+        partition_key = request_parameters.get("PartitionKey")
+        data = request_parameters.get("Data")
+
+        if not stream_name:
+            raise BadRequestException("StreamName")
+
+        if not partition_key:
+            raise BadRequestException("PartitionKey")
+
+        if not data:
+            raise BadRequestException("Data")
+
+    def _create_request_parameters(
+        self, invocation_context: ApiInvocationContext
+    ) -> Dict[str, Any]:
+        request_parameters = invocation_context.integration.get("requestParameters", {})
+        self._validate_required_params(request_parameters)
+
+        variables = {
+            "request": {
+                "header": invocation_context.headers,
+                "querystring": invocation_context.query_params(),
+                "body": invocation_context.data_as_string(),
+                "context": invocation_context.context or {},
+                "stage_variables": invocation_context.stage_variables or {},
+            }
+        }
+
+        if invocation_context.headers.get("Content-Type") == "application/json":
+            variables["request"]["body"] = json.loads(invocation_context.data_as_string())
+        else:
+            # AWS parity no content type still yields a valid response from Kinesis
+            variables["request"]["body"] = try_json(invocation_context.data_as_string())
+
+        # Required parameters
+        payload = {
+            "StreamName": VtlTemplate().render_vtl(request_parameters.get("StreamName"), variables),
+            "Data": VtlTemplate().render_vtl(request_parameters.get("Data"), variables),
+            "PartitionKey": VtlTemplate().render_vtl(
+                request_parameters.get("PartitionKey"), variables
+            ),
+        }
+        # Optional Parameters
+        if "ExplicitHashKey" in request_parameters:
+            payload["ExplicitHashKey"] = VtlTemplate().render_vtl(
+                request_parameters.get("ExplicitHashKey"), variables
+            )
+        if "SequenceNumberForOrdering" in request_parameters:
+            payload["SequenceNumberForOrdering"] = VtlTemplate().render_vtl(
+                request_parameters.get("SequenceNumberForOrdering"), variables
+            )
+        if "Region" in request_parameters:
+            payload["Region"] = VtlTemplate().render_vtl(
+                request_parameters.get("Region"), variables
+            )
```

Comment:
I think we should not pass the `Region` to the Kinesis payload because it's not a supported parameter for `PutRecord` itself, and the Kinesis service will raise an exception.

I know we still need to render it though, so maybe we need to pop it out of the dict after, to properly pass it to the headers creation? Or we don't treat it for now and leaves it as a TODO and a log message indicating it's not supported yet, as creating a multi-region test for it might a bit tricky as this PR is ready otherwise. 

In any case, we should not pass the `Region` parameter directly to the HTTP call.

Reply:
Localstack doesn't support it, you mean? Because is in the docs https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-integrations-aws-services-reference.html#Kinesis-PutRecord

Reply:
LocalStack do support it, but it's not a parameter of PutRecord: https://docs.aws.amazon.com/kinesis/latest/APIReference/API_PutRecord.html#API_PutRecord_RequestParameters

It must mean that API Gateway maps this parameter differently, it does not pass it directly as a parameter to PutRecord but direct it differently. It's also a bit weird as the region should be part of the StreamARN. I think it's fine to leave it as a TODO for now, we can tackle it if we get a report about it. 

Sorry for comment, but at least we avoid the issue if it happens. Thanks for addressing it!

Reply:
Because you can use `StreamName` and not `StreamArn` maybe the region plays a role on that. I don't know.